delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/resources/Resources.scala

```diff
@@ -293,72 +316,104 @@ object Resources {
         case _ => IO.raiseError(ResourceAlreadyExists(c.id, c.project))
       }
 
-    def update(c: UpdateResource) =
+    def validateStateResourceExists(c: ModifyCommand) = {
       state match {
         case None                                                          =>
           IO.raiseError(ResourceNotFound(c.id, c.project, c.schemaOpt))
         case Some(s) if s.rev != c.rev                                     =>
           IO.raiseError(IncorrectRev(c.rev, s.rev))
-        case Some(s) if s.deprecated                                       =>
-          IO.raiseError(ResourceIsDeprecated(c.id))
         case Some(s) if c.schemaOpt.exists(cur => cur.iri != s.schema.iri) =>
           IO.raiseError(UnexpectedResourceSchema(s.id, c.schemaOpt.get, s.schema))
-        case Some(s)                                                       =>
-          // format: off
-          for {
-            (schemaRev, schemaProject) <- validate(s.project, c.schemaOpt.getOrElse(s.schema), c.caller, c.id, c.expanded)
-            types                       = c.expanded.cursor.getTypes.getOrElse(Set.empty)
-            time                       <- IOUtils.instant
-          } yield ResourceUpdated(c.id, c.project, schemaRev, schemaProject, types, c.source, c.compacted, c.expanded, s.rev + 1, time, c.subject)
-        // format: on
-
+        case Some(s)                                                       => IO.now(s)
       }
+    }
 
-    def tag(c: TagResource) =
-      state match {
-        case None                                                          =>
-          IO.raiseError(ResourceNotFound(c.id, c.project, c.schemaOpt))
-        case Some(s) if s.rev != c.rev                                     =>
-          IO.raiseError(IncorrectRev(c.rev, s.rev))
-        case Some(s) if c.schemaOpt.exists(cur => cur.iri != s.schema.iri) =>
-          IO.raiseError(UnexpectedResourceSchema(s.id, c.schemaOpt.get, s.schema))
-        case Some(s) if c.targetRev <= 0 || c.targetRev > s.rev            =>
-          IO.raiseError(RevisionNotFound(c.targetRev, s.rev))
-        case Some(s)                                                       =>
-          IOUtils.instant.map(ResourceTagAdded(c.id, c.project, s.types, c.targetRev, c.tag, s.rev + 1, _, c.subject))
+    def validateStateResourceCanBeModified(c: ModifyCommand) = {
+      validateStateResourceExists(c).flatMap { s =>
+        IO.raiseWhen(s.deprecated)(ResourceIsDeprecated(c.id)).as(s)
+      }
+    }
 
+    def validateStateTagExistsOnResource(c: ModifyCommand, tag: UserTag) = {
```

Comment:
As this one is not reused, what about including it directly in `deleteTag` ?
We would get something like:
```scala
for {
        s     <- resourceExists(c, c.tag)
        _      <- IO.raiseWhen(!s.tags.contains(tag))(TagNotFound(tag))
        time <- IOUtils.instant
      } yield ResourceTagDeleted(c.id, c.project, s.types, c.tag, s.rev + 1, time, c.subject)
```

Reply:
To me it is more readable when the validation is both done in a single method, and named (the method name)